bls/src/main/java/tech/pegasys/teku/bls/impl/blst/BlstPublicKey.java

```diff
@@ -47,33 +45,17 @@ public void forceValidation() {}
         public Bytes48 toBytesCompressed() {
           return INFINITY_COMPRESSED_BYTES;
         }
-
-        @Override
-        public Bytes toBytesUncompressed() {
-          throw new UnsupportedOperationException();
-        }
       };
 
-  public static BlstPublicKey fromBytesUncompressed(Bytes uncompressed) {
-    checkArgument(uncompressed.size() == UNCOMPRESSED_PK_LENGTH);
-    p1_affine ecPoint = new p1_affine();
-    if (blst.p1_deserialize(ecPoint, uncompressed.toArrayUnsafe()) == BLST_ERROR.BLST_SUCCESS) {
-      return new BlstPublicKey(ecPoint);
-    } else {
-      ecPoint.delete();
-      throw new IllegalArgumentException("Invalid PublicKey bytes: " + uncompressed);
-    }
-  }
-
   public static BlstPublicKey fromBytes(Bytes48 compressed) {
     if (compressed.equals(INFINITY_COMPRESSED_BYTES)) {
       return INFINITY;
     }
-    p1_affine ecPoint = new p1_affine();
-    if (blst.p1_uncompress(ecPoint, compressed.toArrayUnsafe()) == BLST_ERROR.BLST_SUCCESS) {
+
+    try {
+      P1_Affine ecPoint = new P1_Affine(compressed.toArrayUnsafe());
       return new BlstPublicKey(ecPoint);
-    } else {
-      ecPoint.delete();
+    } catch (RuntimeException err) {
```

Comment:
It seemed like to create a public key, you would create the `p1_affine` point and then use the `blst.p1_uncompress` method that would return an error in case something was wrong, but now the way to create a public key is just to call the `P1_Affine` constructor and it throws a RuntimeException in case something is wrong, so we do a try catch instead of the if/else here and check for a Runtime exception.

Reply:
Makes sense. Done.